src/main/java/com/yugabyte/sample/apps/MultiTableMultiIndexInserts.java

```diff
@@ -0,0 +1,449 @@
+    valuesPerRow.put("mv", 2);
+
+    baseTableNames.add("a1");
+    baseTableNames.add("a1a");
+    baseTableNames.add("a2");
+    baseTableNames.add("d");
+    baseTableNames.add("a3");
+    baseTableNames.add("a3a");
+    baseTableNames.add("m");
+    baseTableNames.add("mv");
+
+    List<String> a1Indexes = new LinkedList<>();
+    a1Indexes.add("t");
+    a1Indexes.add("f_id_1");
+    a1Indexes.add("f_id_2");
+    indexes.put("a1", a1Indexes);
+
+    List<String> a1aIndexes = new LinkedList<>();
+    a1aIndexes.add("t");
+    a1aIndexes.add("id");
+    a1aIndexes.add("a2_id");
+    indexes.put("a1a", a1aIndexes);
+
+    List<String> a2Indexes = new LinkedList<>();
+    a2Indexes.add("t");
+    indexes.put("a2", a2Indexes);
+
+    List<String> a3Indexes = new LinkedList<>();
+    a3Indexes.add("t");
+    indexes.put("a3", a3Indexes);
+
+    List<String> a3aIndexes = new LinkedList<>();
+    a3aIndexes.add("t");
+    a3aIndexes.add("a3_id");
+    a3aIndexes.add("a2_id");
+    indexes.put("a3a", a3aIndexes);
+
+    List<String> dIndexes = new LinkedList<>();
+    dIndexes.add("t");
+    dIndexes.add("mv_id");
+    indexes.put("d", dIndexes);
+
+    List<String> mIndexes = new LinkedList<>();
+    mIndexes.add("m_id");
+    indexes.put("m", mIndexes);
+  }
+
+  private String getTableName(String baseTableName) { return "Schema1_" + baseTableName; }
+
+  private String getIndexName(String indexName, String baseTableName) {
+      return "Schema1_" + indexName + "_" + baseTableName + "_idx";
+  }
+
+  private String getValuesString(int valuesPerRow) {
+    StringBuilder builder = new StringBuilder();
+    builder.append("(");
+    for (int j = 0; j < valuesPerRow; ++j) {
+      if (j == valuesPerRow - 1) {
+        builder.append("?");
+      } else {
+        builder.append("?, ");
+      }
+    }
+    builder.append(")");
+    return builder.toString();
+  }
+  private String getA2ValuesString() { return "(?, ?::json, ?, ?, ?, ?, ?, ?, ?, ?)"; }
+
+  private PreparedStatement getPreparedInsert(Connection conn, String baseTableName) throws Exception {
+    String tableName = getTableName(baseTableName);
+    String valuesString;
+    if (baseTableName.equals("a2")) {
+      valuesString = getA2ValuesString();
+    } else {
+      valuesString = getValuesString(valuesPerRow.get(baseTableName));
+    }
+    return conn.prepareStatement(String.format("INSERT INTO %s VALUES %s;", tableName, valuesString));
+  }
+
+  @Override
+  public void initializeConnectionsAndStatements(int numThreads) {
+    for (int i = 0; i < numThreads; ++i) {
+      try {
+        Connection conn = getPostgresConnection();
+        Map<String, PreparedStatement> m = new HashMap<>();
+
+        for (String baseTableName : baseTableNames) {
+          m.put(getTableName(baseTableName), getPreparedInsert(conn, baseTableName));
+        }
+        threadStates.put(i, new PerThreadState(conn, m));
+      } catch (Exception e) {
+        LOG.error("...", e);
```

Comment:
I did not literally mean `"..."`, I meant the log message that you previously had here.

Reply:
Ooops.. I thought you wanted that.
Even I was super surprised that you asked me to do that.. :)